File: js/demo/objects/pause-menu.js
'use strict';
import { Drawable, GameObject, override } from '../../engine';
import Menu from './menu';
import MainMenu from '../rooms/main-menu';

class PauseMenu extends Menu {
  @override
  init() {
    this.opts = {
      ['Resume']: () => {
        super.game.room.close();
      },
      ['Main Menu']: () => {
        super.game.room.close();
        super.game.room.goto(MainMenu);
      }
    };
  }
}

export default PauseMenu;

File: c++/src/demo/objects/pause-menu.cc
#include "pause-menu.h"
#include "../rooms/main-menu.h"
#include <iostream>

namespace Demo {
    PauseMenu::PauseMenu() : Menu {std::vector<Menu::Option>{
        {"Resume", [this] () { game().room_close(); } },
        {"Main Menu", [this] () {
            game().room_close();
            game().room_goto<RmMainMenu>();
        } }
    }} {}
}
